doc/pveclibmaindox.h

```diff
@@ -414,6 +416,93 @@
 *  provides inspiration for SIMD versions of; count leading zeros,
 *  population count, parity, etc.
 *
+*  \subsubsection mainpage_endian_issues_1_1 General Endian Issues
+*
+*  For POWER8, IBM made the explicit decision to support Little Endian
+*  (<B>LE</B>) data format in the Linux ecosystem.  The goal was to
+*  enhance application code portability across Linux platforms.
+*  This goal was integrated into the OpenPOWER ELF V2
+*  Application Binary Interface <B>ABI</B> specification.
+*
+*  The POWER8 processor architecturally supports an <I>Endian Mode</I>
+*  and supports both BE and LE storage access in hardware.
+*  However register to register operations do not change
+*  (are not effected by endian mode).  The ABI extends the LE
+*  storage format to vector register (logical)
+*  element numbering.
+*
+*  This has no effect for most altivec.h
+*  operations where the input elements and the results "stay in their
+*  lanes".  For operations of the form (T[n] = A[n] op B[n]),
+*  it does not matter if elements are numbered [0, 1, 2, 3] or
+*  [3, 2, 1, 0].
+*
+*  But there are cases where element renumbering can change the
+*  results.  Changing element numbering does change the even / odd
+*  relationship for merge and integer multiply.  For <B>LE</B>
+*  targets, operations accessing even vector elements are
+*  implemented using the equivalent odd instruction (and visa versa)
+*  and inputs are swapped.  Similarly for high and low merges.
+*  Inputs are also swapped for Pack, Unpack, and Permute operations
+*  and the permute select vector is inverted.
+*  The above is just a sampling of a larger list <I>LE transforms</I>.
+*  The OpenPOWER ABI specification provides a helpful table of
+*  <a href="http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655244_90667.html"
+*  >Endian-Sensitive Operations</a>.
+*
+*  \note This means that the vector built-ins provided by altivec.h
+*  may not generate the instructions you expect.
+*
+*  This does matter when doing extended precision arithmetic.
+*  Here we need to maintain most-to-least significant byte order and
+*  align "digit" columns for summing partial products.  Many of these
+*  operation where defined long before Little Endian was seriously
+*  considered and are decidedly Big Endian in register format.
+*  Basically any operation where the element
+*  size changes from input to output is suspect for <B>LE</B>
+*  targets.
```

Comment:
Not sure what this means. Do you mean like halfword to word?  This is one category, but not the only one.  Shifts, for example.  And you just mentioned the even/odd operations.

Reply:
The list is almost endless mul, pack, unpack, convert, ..... A complete list would go on for pages (and is covered in the element specific headers). Anyone who wants to really understand this will have to study all the previous references and the code. There is no way to cover this in a single paragraph. All I can do here is provide fair warning.

How about?:
*  Basically, any operation where the element changes size 
*  (truncated, extended, converted, subseted) from input to output
*  is suspect for <B>LE</B> targets.

Any one who just wants their code to work, should use pveclib!